components/invitation_modal/invitation_modal.jsx

```diff
@@ -0,0 +1,188 @@
+import ConfirmModal from 'components/confirm_modal.jsx';
+import RootPortal from 'components/root_portal';
+
+import InvitationModalInitialStep from './invitation_modal_initial_step.jsx';
+import InvitationModalMembersStep from './invitation_modal_members_step.jsx';
+import InvitationModalGuestsStep from './invitation_modal_guests_step.jsx';
+import InvitationModalConfirmStep from './invitation_modal_confirm_step.jsx';
+
+import './invitation_modal.scss';
+
+const STEPS_INITIAL = 'initial';
+const STEPS_INVITE_MEMBERS = 'members';
+const STEPS_INVITE_GUESTS = 'guests';
+const STEPS_INVITE_CONFIRM = 'confirm';
+
+export default class InvitationModal extends React.Component {
+    static propTypes = {
+        show: PropTypes.bool,
+        currentTeam: PropTypes.object.isRequired,
+        invitableChannels: PropTypes.array.isRequired,
+        canInviteGuests: PropTypes.bool.isRequired,
+        canAddUsers: PropTypes.bool.isRequired,
+        actions: PropTypes.shape({
+            closeModal: PropTypes.func.isRequired,
+            sendGuestsInvites: PropTypes.func.isRequired,
+            sendMembersInvites: PropTypes.func.isRequired,
+            searchProfiles: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        let step = STEPS_INITIAL;
+        if (!props.canInviteGuests) {
+            step = STEPS_INVITE_MEMBERS;
+        }
+
+        if (!props.canAddUsers) {
+            step = STEPS_INVITE_GUESTS;
+        }
+
+        this.state = {
+            step,
+            confirmModal: false,
+            hasChanges: false,
+            invitesType: 'member',
+            invitesSent: [],
+            invitesNotSent: [],
+        };
+    }
+
+    goToInitialStep = () => {
+        this.setState({step: STEPS_INITIAL, hasChanges: false});
+    }
+
+    goToMembers = () => {
+        this.setState({step: STEPS_INVITE_MEMBERS, hasChanges: false, invitesSent: [], invitesNotSent: [], invitesType: 'member'});
+    }
+
+    goToGuests = () => {
+        this.setState({step: STEPS_INVITE_GUESTS, hasChanges: false, invitesSent: [], invitesNotSent: [], invitesType: 'guest'});
+        this.setState({});
+    }
+
+    onEdit = () => {
+        this.setState({hasChanges: true});
+    }
+
+    close = async () => {
+        if (this.state.hasChanges) {
+            this.setState({confirmModal: true});
+        } else {
+            this.props.actions.closeModal();
+        }
+    }
+
+    confirmClose = () => {
+        this.props.actions.closeModal();
+        this.setState({confirmModal: false});
+    }
+
+    cancelConfirm = () => {
+        this.setState({confirmModal: false});
+    }
+
+    onMembersSubmit = async (users, emails) => {
+        const invites = await this.props.actions.sendMembersInvites(this.props.currentTeam.id, users, emails);
```

Comment:
Is this handling an error response?

Reply:
Actually, here all the error handling is done at the action level, the remote handle the errors inside and always give the proper return so... here extra error handling is not needed.